code/src/main/webapp/JavaScript/script.js

```diff
@@ -170,24 +170,18 @@ function attachSignin(element, negativeUser) {
     document.getElementById('name').innerText = "Signed in: " + googleUser.getBasicProfile().getName();
 
     const idToken = googleUser.getAuthResponse().id_token;
+    localStorage.setItem('idToken', idToken.toString());
     const params = new URLSearchParams()
     params.append('idToken', idToken);
-    const request = new Request('/authenticate', {method: 'POST', body: params});
-    fetch(request).then(() => {
-      if (negativeUser) {
-        PAGE_CONTROLLER.show('notification')
-      }
-    });
+    fetch(new Request('/authentication-test', {method: 'POST', body: params}))
+      .then(response => response.text())
+      .then(url => {if (url == "Error") {
+        alert("something went wrong, please try again");
```

Comment:
We might want to have a more informative alert than this. You're showing this to the actual user?